bukkit/rpk-unconsciousness-bukkit/src/main/kotlin/com/rpkit/unconsciousness/bukkit/unconsciousness/RPKUnconsciousnessServiceImpl.kt

```diff
@@ -24,31 +24,33 @@ import org.bukkit.potion.PotionEffect
 import org.bukkit.potion.PotionEffectType
 import java.time.LocalDateTime
 import java.time.temporal.ChronoUnit
+import java.util.concurrent.CompletableFuture
 
 
 class RPKUnconsciousnessServiceImpl(override val plugin: RPKUnconsciousnessBukkit) : RPKUnconsciousnessService {
 
-    override fun isUnconscious(character: RPKCharacter): Boolean {
+    override fun isUnconscious(character: RPKCharacter): CompletableFuture<Boolean> {
         val unconsciousStateTable = plugin.database.getTable(RPKUnconsciousStateTable::class.java)
-        val unconsciousState = unconsciousStateTable.get(character)
-        return if (unconsciousState == null) {
-            false
-        } else {
-            unconsciousState.deathTime.plus(plugin.config.getLong("unconscious-time"), ChronoUnit.MILLIS) > LocalDateTime.now()
+        return unconsciousStateTable[character].thenApply { unconsciousState ->
+            return@thenApply if (unconsciousState == null) {
+                false
+            } else {
+                unconsciousState.deathTime.plus(plugin.config.getLong("unconscious-time"), ChronoUnit.MILLIS) > LocalDateTime.now()
+            }
         }
     }
 
-    override fun setUnconscious(character: RPKCharacter, unconscious: Boolean) {
+    override fun setUnconscious(character: RPKCharacter, unconscious: Boolean): CompletableFuture<Void> {
```

Comment:
I would wrap the entire body of this method in a CompletableFuture.runAsync call. This allows you to do joins freely as you're not blocking the calling thread.
If you do this, add an `isAsync` parameter to the RPKBukkitUnconsciousnessStateChangeEvent and pass that to RPKBukkitEvent, as when events are fired async they need to be flagged as such to Bukkit/Spigot.
Firing the events async has the added benefit of allowing async calls in listeners for them.